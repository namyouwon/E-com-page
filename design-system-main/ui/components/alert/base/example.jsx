// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
