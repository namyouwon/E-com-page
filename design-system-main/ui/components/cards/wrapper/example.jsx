// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Card, { CardHeader, CardBody, CardFooter } from '../';
import Tabs from '../../tabs/index.react';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <div className="slds-card-wrapper">
        <Card>
          <CardHeader
            title="Contacts"
            href
            symbol="contact"
            action={
              <button className="slds-button slds-button_neutral">New</button>
            }
          />
          <CardBody>
            This is a card inside an `slds-card-wrapper` to show how styling is
            removed when cards are nested inside.
          </CardBody>
          <CardFooter>
            View All <span className="slds-assistive-text">Contacts</span>
          </CardFooter>
        </Card>
        <Card hasCardBoundary>
          <CardHeader
            title="Contacts"
            href
            symbol="contact"
            action={
              <button className="slds-button slds-button_neutral">New</button>
            }
          />
          <CardBody hasPadding>
            This is a card inside an `slds-card-wrapper` to illustrate how
            `slds-card_boundary` adds a rounded border when desired.
          </CardBody>
          <CardFooter>
            View All <span className="slds-assistive-text">Contacts</span>
          </CardFooter>
        </Card>
      </div>
    )
  }
];

export let examples = [
  {
    id: 'tabs-wrapper',
    label: 'Card wrapper for tabs',
    element: (
      <div className="slds-card-wrapper">
        <Tabs selectedIndex={0}>
          <Tabs.Item title="Item One" id="tab-default-1">
            <Card>
              <CardHeader
                title="Contacts"
                href
                symbol="contact"
                action={
                  <button className="slds-button slds-button_neutral">
                    New
                  </button>
                }
              />
              <CardBody>
                This is a card inside an `slds-card-wrapper` to show how styling
                is removed when cards are nested inside.
              </CardBody>
              <CardFooter>
                View All <span className="slds-assistive-text">Contacts</span>
              </CardFooter>
            </Card>
            <Card hasCardBoundary>
              <CardHeader
                title="Contacts"
                href
                symbol="contact"
                action={
                  <button className="slds-button slds-button_neutral">
                    New
                  </button>
                }
              />
              <CardBody hasPadding>
                This is a card inside an `slds-card-wrapper` to illustrate how
                `slds-card_boundary` adds a rounded border when desired.
              </CardBody>
              <CardFooter>
                View All <span className="slds-assistive-text">Contacts</span>
              </CardFooter>
            </Card>
          </Tabs.Item>
          <Tabs.Item title="Item Two" id="tab-default-2">
            Item Two Content
          </Tabs.Item>
          <Tabs.Item title="Item Three" id="tab-default-3">
            Item Three Content
          </Tabs.Item>
        </Tabs>
      </div>
    )
  },
  {
    id: 'styling-hooks',
    label: 'Styling Hooks',
    demoStyles: `
      --slds-c-card-color-background: #f3f3f3;
      --slds-c-card-radius-border: 8px;
      --slds-c-card-shadow: rgba(0,0,0,.15) 0 4px 8px, rgba(0,0,0,.05) 0 2px 4px;
      --slds-c-card-spacing-block-start: 1.5rem;
      --slds-c-card-spacing-block-end: 0.5rem;
      --slds-c-card-title-font-size: 1.5rem;
      --slds-c-card-header-spacing-block-start: 0;
      --slds-c-card-header-spacing-inline-start: 1.5rem;
      --slds-c-card-header-spacing-inline-end: 1.5rem;
      --slds-c-card-body-spacing-inline-start: 1.5rem;
      --slds-c-card-body-spacing-inline-end: 1.5rem;
      --slds-c-card-body-spacing-block-end: 1rem;
      --slds-c-card-footer-spacing-block-start: 0.5rem;
      --slds-c-card-footer-spacing-block-end: 0;
      --slds-c-card-footer-spacing-inline-start: 1.5rem;
      --slds-c-card-footer-spacing-inline-end: 1.5rem;
      --slds-c-card-footer-font-size: 0.75rem;
      --slds-c-card-footer-text-align: right;
    `,
    storybookStyles: true,
    element: (
      <div className="slds-card-wrapper">
        <Tabs selectedIndex={0}>
          <Tabs.Item title="Item One" id="tab-default-1">
            <Card>
              <CardHeader
                title="Contacts"
                href
                symbol="contact"
                action={
                  <button className="slds-button slds-button_neutral">
                    New
                  </button>
                }
              />
              <CardBody>
                This is a card inside an `slds-card-wrapper` to show how styling
                is removed when cards are nested inside.
              </CardBody>
              <CardFooter>
                View All <span className="slds-assistive-text">Contacts</span>
              </CardFooter>
            </Card>
            <Card hasCardBoundary>
              <CardHeader
                title="Contacts"
                href
                symbol="contact"
                action={
                  <button className="slds-button slds-button_neutral">
                    New
                  </button>
                }
              />
              <CardBody hasPadding>
                This is a card inside an `slds-card-wrapper` to illustrate how
                `slds-card_boundary` adds a rounded border when desired.
              </CardBody>
              <CardFooter>
                View All <span className="slds-assistive-text">Contacts</span>
              </CardFooter>
            </Card>
          </Tabs.Item>
          <Tabs.Item title="Item Two" id="tab-default-2">
            Item Two Content
          </Tabs.Item>
          <Tabs.Item title="Item Three" id="tab-default-3">
            Item Three Content
          </Tabs.Item>
        </Tabs>
      </div>
    )
  }
];
