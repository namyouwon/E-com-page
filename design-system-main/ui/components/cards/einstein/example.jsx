// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Card, { CardBody, CardFooter } from '../';
import { EinsteinHeader } from '../../einstein-header/base/';

export const EinsteinCard = props => (
  <Card>
    <EinsteinHeader
      className="slds-card__header"
      headerTitle="Einstein (10+)"
      hasLink
      actions={props.hasActions}
      symbol={props.hasIcon && 'salesforce1'}
    />
    <CardBody hasPadding>Card Body</CardBody>
    <CardFooter>
      View All{' '}
      <span className="slds-assistive-text">Einstein Recommendations</span>
    </CardFooter>
  </Card>
);

export default <EinsteinCard />;

export let states = [
  {
    id: 'einstein-with-icon',
    label: 'Einstein card - with icon',
    element: <EinsteinCard hasIcon />
  },
  {
    id: 'einstein-with-icon-and-actions',
    label: 'Einstein card - with icon and actions',
    element: <EinsteinCard hasIcon hasActions />
  }
];
