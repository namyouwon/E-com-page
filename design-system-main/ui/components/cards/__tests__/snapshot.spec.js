/* eslint-env jest */
import React from 'react';
import Card, { CardHeader, CardBody, CardFooter } from '../';
import { EinsteinCard } from '../einstein/example';

import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

describe('Card Article', () => {
  it('renders a basic card article', () =>
    matchesMarkup(<Card>Anything can go into the card body</Card>));
  it('renders a basic card article with boundary', () =>
    matchesMarkup(
      <Card hasCardBoundary>Anything can go into the card body</Card>
    ));
});

describe('Card Header', () => {
  it('renders a simple card header', () =>
    matchesMarkup(
      <CardHeader
        title="Card Header"
        href="#"
        onClick={e => e.preventDefault()}
      />
    ));
  it('renders a card header with icon', () =>
    matchesMarkup(
      <CardHeader
        title="Card Header"
        href="#"
        onClick={e => e.preventDefault()}
        symbol="contact"
      />
    ));
  it('renders a card header with action', () =>
    matchesMarkup(
      <CardHeader
        title="Card Header"
        href="#"
        onClick={e => e.preventDefault()}
        action={
          <button className="slds-button slds-button_neutral">New</button>
        }
      />
    ));
  it('renders a card header with icon and action', () =>
    matchesMarkup(
      <CardHeader
        title="Card Header"
        href="#"
        onClick={e => e.preventDefault()}
        symbol="contact"
        action={
          <button className="slds-button slds-button_neutral">New</button>
        }
      />
    ));
});

describe('Card Body', () => {
  it('renders an empty card body', () => matchesMarkup(<CardBody />));
  it('renders a card body with content', () =>
    matchesMarkup(<CardBody>Anything can go into the card body</CardBody>));
  it('renders a card body with padding', () =>
    matchesMarkup(
      <CardBody hasPadding>Anything can go into the card body</CardBody>
    ));
});

describe('Card Footer', () => {
  it('renders an empty card footer', () => matchesMarkup(<CardFooter />));
  it('renders a card footer with content', () =>
    matchesMarkup(
      <CardFooter>
        View All <span className="slds-assistive-text">Accounts</span>
      </CardFooter>
    ));
});

describe('Einstein Card', () => {
  it('renders a base einstein card', () => matchesMarkup(<EinsteinCard />));
  it('renders an einstein card with icon', () =>
    matchesMarkup(<EinsteinCard hasIcon />));
  it('renders an einstein card with actions', () =>
    matchesMarkup(<EinsteinCard hasActions />));
  it('renders an einstein card with icon and actions', () =>
    matchesMarkup(<EinsteinCard hasIcon hasActions />));
});

describe('Card', () => {
  it('renders a base card empty', () =>
    matchesMarkup(
      <Card>
        <CardHeader
          title="Card Header"
          href="#"
          onClick={e => e.preventDefault()}
          symbol="contact"
        />
        <CardBody />
        <CardFooter />
      </Card>
    ));
  it('renders a base card without a footer', () =>
    matchesMarkup(
      <Card>
        <CardHeader
          title="Card Header"
          href="#"
          onClick={e => e.preventDefault()}
          symbol="contact"
        />
        <CardBody>Anything can go into the card body</CardBody>
      </Card>
    ));
  it('renders a base card with empty footer', () =>
    matchesMarkup(
      <Card>
        <CardHeader
          title="Card Header"
          href="#"
          onClick={e => e.preventDefault()}
          symbol="contact"
        />
        <CardBody>Anything can go into the card body</CardBody>
        <CardFooter />
      </Card>
    ));
  it('renders a base card with content in footer', () =>
    matchesMarkup(
      <Card>
        <CardHeader
          title="Card Header"
          href="#"
          onClick={e => e.preventDefault()}
          symbol="contact"
        />
        <CardBody>Anything can go into the card body</CardBody>
        <CardFooter>
          View All <span className="slds-assistive-text">Contacts</span>
        </CardFooter>
      </Card>
    ));
  it('renders a base card with header visibly hidden', () =>
    matchesMarkup(
      <Card>
        <CardHeader
          title="Card Header"
          href="#"
          onClick={e => e.preventDefault()}
          symbol="contact"
          className="slds-assistive-text"
        />
        <CardBody>Anything can go into the card body</CardBody>
        <CardFooter>
          View All <span className="slds-assistive-text">Contacts</span>
        </CardFooter>
      </Card>
    ));
});
