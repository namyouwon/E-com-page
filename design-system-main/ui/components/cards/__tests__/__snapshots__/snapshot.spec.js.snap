// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Article renders a basic card article 1`] = `
<article
  className="slds-card"
>
  Anything can go into the card body
</article>
`;

exports[`Card Article renders a basic card article with boundary 1`] = `
<article
  className="slds-card slds-card_boundary"
>
  Anything can go into the card body
</article>
`;

exports[`Card Body renders a card body with content 1`] = `
<div
  className="slds-card__body"
>
  Anything can go into the card body
</div>
`;

exports[`Card Body renders a card body with padding 1`] = `
<div
  className="slds-card__body slds-card__body_inner"
>
  Anything can go into the card body
</div>
`;

exports[`Card Body renders an empty card body 1`] = `
<div
  className="slds-card__body"
/>
`;

exports[`Card Footer renders a card footer with content 1`] = `
<footer
  className="slds-card__footer"
>
  <a
    className="slds-card__footer-action"
    href="#"
    onClick={[Function]}
  >
    View All 
    <span
      className="slds-assistive-text"
    >
      Accounts
    </span>
  </a>
</footer>
`;

exports[`Card Footer renders an empty card footer 1`] = `
<footer
  className="slds-card__footer"
/>
`;

exports[`Card Header renders a card header with action 1`] = `
<div
  className="slds-card__header slds-grid"
>
  <header
    className="slds-media slds-media_center slds-has-flexi-truncate"
  >
    <div
      className="slds-media__body"
    >
      <h2
        className="slds-card__header-title"
      >
        <a
          className="slds-card__header-link slds-truncate"
          href="#"
          onClick={[Function]}
          title="Card Header"
        >
          <span>
            Card Header
          </span>
        </a>
      </h2>
    </div>
    <div
      className="slds-no-flex"
    >
      <button
        className="slds-button slds-button_neutral"
      >
        New
      </button>
    </div>
  </header>
</div>
`;

exports[`Card Header renders a card header with icon 1`] = `
<div
  className="slds-card__header slds-grid"
>
  <header
    className="slds-media slds-media_center slds-has-flexi-truncate"
  >
    <div
      className="slds-media__figure"
    >
      <span
        className="slds-icon_container slds-icon-standard-contact"
        title="contact"
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon_small"
        >
          <use
            xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          contact
        </span>
      </span>
    </div>
    <div
      className="slds-media__body"
    >
      <h2
        className="slds-card__header-title"
      >
        <a
          className="slds-card__header-link slds-truncate"
          href="#"
          onClick={[Function]}
          title="Card Header"
        >
          <span>
            Card Header
          </span>
        </a>
      </h2>
    </div>
  </header>
</div>
`;

exports[`Card Header renders a card header with icon and action 1`] = `
<div
  className="slds-card__header slds-grid"
>
  <header
    className="slds-media slds-media_center slds-has-flexi-truncate"
  >
    <div
      className="slds-media__figure"
    >
      <span
        className="slds-icon_container slds-icon-standard-contact"
        title="contact"
      >
        <svg
          aria-hidden={true}
          className="slds-icon slds-icon_small"
        >
          <use
            xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          contact
        </span>
      </span>
    </div>
    <div
      className="slds-media__body"
    >
      <h2
        className="slds-card__header-title"
      >
        <a
          className="slds-card__header-link slds-truncate"
          href="#"
          onClick={[Function]}
          title="Card Header"
        >
          <span>
            Card Header
          </span>
        </a>
      </h2>
    </div>
    <div
      className="slds-no-flex"
    >
      <button
        className="slds-button slds-button_neutral"
      >
        New
      </button>
    </div>
  </header>
</div>
`;

exports[`Card Header renders a simple card header 1`] = `
<div
  className="slds-card__header slds-grid"
>
  <header
    className="slds-media slds-media_center slds-has-flexi-truncate"
  >
    <div
      className="slds-media__body"
    >
      <h2
        className="slds-card__header-title"
      >
        <a
          className="slds-card__header-link slds-truncate"
          href="#"
          onClick={[Function]}
          title="Card Header"
        >
          <span>
            Card Header
          </span>
        </a>
      </h2>
    </div>
  </header>
</div>
`;

exports[`Card renders a base card empty 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-card__header slds-grid"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-standard-contact"
          title="contact"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            contact
          </span>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2
          className="slds-card__header-title"
        >
          <a
            className="slds-card__header-link slds-truncate"
            href="#"
            onClick={[Function]}
            title="Card Header"
          >
            <span>
              Card Header
            </span>
          </a>
        </h2>
      </div>
    </header>
  </div>
  <div
    className="slds-card__body"
  />
  <footer
    className="slds-card__footer"
  />
</article>
`;

exports[`Card renders a base card with content in footer 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-card__header slds-grid"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-standard-contact"
          title="contact"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            contact
          </span>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2
          className="slds-card__header-title"
        >
          <a
            className="slds-card__header-link slds-truncate"
            href="#"
            onClick={[Function]}
            title="Card Header"
          >
            <span>
              Card Header
            </span>
          </a>
        </h2>
      </div>
    </header>
  </div>
  <div
    className="slds-card__body"
  >
    Anything can go into the card body
  </div>
  <footer
    className="slds-card__footer"
  >
    <a
      className="slds-card__footer-action"
      href="#"
      onClick={[Function]}
    >
      View All 
      <span
        className="slds-assistive-text"
      >
        Contacts
      </span>
    </a>
  </footer>
</article>
`;

exports[`Card renders a base card with empty footer 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-card__header slds-grid"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-standard-contact"
          title="contact"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            contact
          </span>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2
          className="slds-card__header-title"
        >
          <a
            className="slds-card__header-link slds-truncate"
            href="#"
            onClick={[Function]}
            title="Card Header"
          >
            <span>
              Card Header
            </span>
          </a>
        </h2>
      </div>
    </header>
  </div>
  <div
    className="slds-card__body"
  >
    Anything can go into the card body
  </div>
  <footer
    className="slds-card__footer"
  />
</article>
`;

exports[`Card renders a base card with header visibly hidden 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-card__header slds-grid slds-assistive-text"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-standard-contact"
          title="contact"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            contact
          </span>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2
          className="slds-card__header-title"
        >
          <a
            className="slds-card__header-link slds-truncate"
            href="#"
            onClick={[Function]}
            title="Card Header"
          >
            <span>
              Card Header
            </span>
          </a>
        </h2>
      </div>
    </header>
  </div>
  <div
    className="slds-card__body"
  >
    Anything can go into the card body
  </div>
  <footer
    className="slds-card__footer"
  >
    <a
      className="slds-card__footer-action"
      href="#"
      onClick={[Function]}
    >
      View All 
      <span
        className="slds-assistive-text"
      >
        Contacts
      </span>
    </a>
  </footer>
</article>
`;

exports[`Card renders a base card without a footer 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-card__header slds-grid"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-media__figure"
      >
        <span
          className="slds-icon_container slds-icon-standard-contact"
          title="contact"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#contact"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            contact
          </span>
        </span>
      </div>
      <div
        className="slds-media__body"
      >
        <h2
          className="slds-card__header-title"
        >
          <a
            className="slds-card__header-link slds-truncate"
            href="#"
            onClick={[Function]}
            title="Card Header"
          >
            <span>
              Card Header
            </span>
          </a>
        </h2>
      </div>
    </header>
  </div>
  <div
    className="slds-card__body"
  >
    Anything can go into the card body
  </div>
</article>
`;

exports[`Einstein Card renders a base einstein card 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-grid slds-einstein-header slds-card__header"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-grid slds-grid_vertical-align-center slds-size_3-of-4 slds-medium-size_2-of-3"
      >
        <div
          className="slds-media__body"
        >
          <h2
            className="slds-truncate"
            title="Einstein (10+)"
          >
            <a
              className="slds-card__header-link"
              href="#"
              onClick={[Function]}
              title="Einstein (10+)"
            >
              <span
                className="slds-text-heading_small"
              >
                Einstein (10+)
              </span>
            </a>
          </h2>
        </div>
      </div>
      <div
        className="slds-einstein-header__figure slds-size_1-of-4 slds-medium-size_1-of-3"
      />
    </header>
  </div>
  <div
    className="slds-card__body slds-card__body_inner"
  >
    Card Body
  </div>
  <footer
    className="slds-card__footer"
  >
    <a
      className="slds-card__footer-action"
      href="#"
      onClick={[Function]}
    >
      View All
       
      <span
        className="slds-assistive-text"
      >
        Einstein Recommendations
      </span>
    </a>
  </footer>
</article>
`;

exports[`Einstein Card renders an einstein card with actions 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-grid slds-einstein-header slds-card__header"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-grid slds-grid_vertical-align-center slds-size_3-of-4 slds-medium-size_2-of-3"
      >
        <div
          className="slds-media__body"
        >
          <h2
            className="slds-truncate"
            title="Einstein (10+)"
          >
            <a
              className="slds-card__header-link"
              href="#"
              onClick={[Function]}
              title="Einstein (10+)"
            >
              <span
                className="slds-text-heading_small"
              >
                Einstein (10+)
              </span>
            </a>
          </h2>
        </div>
      </div>
      <div
        className="slds-einstein-header__figure slds-size_1-of-4 slds-medium-size_1-of-3"
      />
    </header>
    <div
      className="slds-einstein-header__actions"
    >
      <div
        className="slds-button-group"
        role="group"
      >
        <button
          aria-pressed="false"
          className="slds-button slds-button_icon slds-button_icon-border-filled"
          onClick={[Function]}
          title="Email"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#email"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Email
          </span>
        </button>
        <button
          aria-pressed="false"
          className="slds-button slds-button_icon slds-button_icon-border-filled"
          onClick={[Function]}
          title="Edit"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#edit"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Edit
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="slds-card__body slds-card__body_inner"
  >
    Card Body
  </div>
  <footer
    className="slds-card__footer"
  >
    <a
      className="slds-card__footer-action"
      href="#"
      onClick={[Function]}
    >
      View All
       
      <span
        className="slds-assistive-text"
      >
        Einstein Recommendations
      </span>
    </a>
  </footer>
</article>
`;

exports[`Einstein Card renders an einstein card with icon 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-grid slds-einstein-header slds-card__header"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-grid slds-grid_vertical-align-center slds-size_3-of-4 slds-medium-size_2-of-3"
      >
        <div
          className="slds-media__figure"
        >
          <span
            className="slds-icon_container slds-icon-utility-salesforce1"
          >
            <svg
              aria-hidden={true}
              className="slds-icon slds-icon_small slds-icon-text-default"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
              />
            </svg>
          </span>
        </div>
        <div
          className="slds-media__body"
        >
          <h2
            className="slds-truncate"
            title="Einstein (10+)"
          >
            <a
              className="slds-card__header-link"
              href="#"
              onClick={[Function]}
              title="Einstein (10+)"
            >
              <span
                className="slds-text-heading_small"
              >
                Einstein (10+)
              </span>
            </a>
          </h2>
        </div>
      </div>
      <div
        className="slds-einstein-header__figure slds-size_1-of-4 slds-medium-size_1-of-3"
      />
    </header>
  </div>
  <div
    className="slds-card__body slds-card__body_inner"
  >
    Card Body
  </div>
  <footer
    className="slds-card__footer"
  >
    <a
      className="slds-card__footer-action"
      href="#"
      onClick={[Function]}
    >
      View All
       
      <span
        className="slds-assistive-text"
      >
        Einstein Recommendations
      </span>
    </a>
  </footer>
</article>
`;

exports[`Einstein Card renders an einstein card with icon and actions 1`] = `
<article
  className="slds-card"
>
  <div
    className="slds-grid slds-einstein-header slds-card__header"
  >
    <header
      className="slds-media slds-media_center slds-has-flexi-truncate"
    >
      <div
        className="slds-grid slds-grid_vertical-align-center slds-size_3-of-4 slds-medium-size_2-of-3"
      >
        <div
          className="slds-media__figure"
        >
          <span
            className="slds-icon_container slds-icon-utility-salesforce1"
          >
            <svg
              aria-hidden={true}
              className="slds-icon slds-icon_small slds-icon-text-default"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
              />
            </svg>
          </span>
        </div>
        <div
          className="slds-media__body"
        >
          <h2
            className="slds-truncate"
            title="Einstein (10+)"
          >
            <a
              className="slds-card__header-link"
              href="#"
              onClick={[Function]}
              title="Einstein (10+)"
            >
              <span
                className="slds-text-heading_small"
              >
                Einstein (10+)
              </span>
            </a>
          </h2>
        </div>
      </div>
      <div
        className="slds-einstein-header__figure slds-size_1-of-4 slds-medium-size_1-of-3"
      />
    </header>
    <div
      className="slds-einstein-header__actions"
    >
      <div
        className="slds-button-group"
        role="group"
      >
        <button
          aria-pressed="false"
          className="slds-button slds-button_icon slds-button_icon-border-filled"
          onClick={[Function]}
          title="Email"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#email"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Email
          </span>
        </button>
        <button
          aria-pressed="false"
          className="slds-button slds-button_icon slds-button_icon-border-filled"
          onClick={[Function]}
          title="Edit"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#edit"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Edit
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="slds-card__body slds-card__body_inner"
  >
    Card Body
  </div>
  <footer
    className="slds-card__footer"
  >
    <a
      className="slds-card__footer-action"
      href="#"
      onClick={[Function]}
    >
      View All
       
      <span
        className="slds-assistive-text"
      >
        Einstein Recommendations
      </span>
    </a>
  </footer>
</article>
`;
